viewer/packages/360-images/src/icons/IconCollection.ts

```diff
@@ -0,0 +1,211 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+
+import { Camera, CanvasTexture, Matrix4, Texture, Vector3, WebGLRenderer } from 'three';
+import { BeforeSceneRenderedDelegate, EventTrigger, SceneHandler } from '@reveal/utilities';
+import { Image360Icon } from './Image360Icon';
+import { InstancedIconSprite } from './InstancedIconSprite';
+import { IconOctree } from './IconOctree';
+
+export class Image360CollectionIcons {
+  private readonly MIN_PIXEL_SIZE = 16;
+  private readonly MAX_PIXEL_SIZE = 64;
+  private readonly _sceneHandler: SceneHandler;
+  private readonly _hoverIconTexture: CanvasTexture;
+  private readonly _onRenderTrigger: EventTrigger<(renderer: WebGLRenderer, camera: Camera) => void>;
+  private readonly _sharedTexture: Texture;
+  private readonly _icons: Image360Icon[];
+  private readonly _iconsSprite: InstancedIconSprite;
+  private readonly _clusterSprites: InstancedIconSprite;
+
+  get icons(): Image360Icon[] {
+    return this._icons;
+  }
+
+  constructor(
+    points: Vector3[],
+    sceneHandler: SceneHandler,
+    onBeforeSceneRendered: EventTrigger<BeforeSceneRenderedDelegate>
+  ) {
+    const sharedTexture = this.createOuterRingsTexture();
+    this._onRenderTrigger = new EventTrigger();
+    const iconsSprites = new InstancedIconSprite(
+      points.length,
+      sharedTexture,
+      this.MIN_PIXEL_SIZE,
+      this.MAX_PIXEL_SIZE,
+      0.5
```

Comment:
Would be good to have the radius as a `const` variables in here as well for clustered sprites